docs/contributing/our-backward-compatibility-promise.md

```diff
@@ -0,0 +1,71 @@
+# Our Backward Compatibility Promise
+
+Smooth and safe upgrades of your own e-commerce project are very important to us.
+In the same time, we need to be able to improve Shopsys Framework for you by adding functionality, enhancing or simplifying current functions and fixing bugs.
+This is why we've written this page explaining what amount of backward compatibility should be expected and how we plan to make changes in the future.
+
+Shopsys Framework is built on the shoulders of giants so we've based our BC promise on the [Symfony Backward Compatibility Promise](https://symfony.com/doc/3.4/contributing/code/bc.html).
+Exceptions from adhering to Symfony's promise and clarifications for non-PHP source codes can be found below.
+
+*Note: This BC promise becomes effective with the release first non-beta release (`7.0.0`).*
+
+## Releases and Versioning
+This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html), which means it's release versions are in format `MAJOR.MINOR.PATCH`:
+
+- `MAJOR` version may contain incompatible changes
+- `MINOR` version may add new functionality in a backward-compatible manner
+- `PATCH` version contains only backward-compatible bug fixes
+
+Released versions will be always marked using git tags with `v` prefix (eg. `v7.0.0`).
+Once created, a git tag marking a release will never be removed or edited.
+
+*Note: Pre-release versions may introduce incompatible changes and can be used to try out the new functions and changes.
+Pre-release version format is `MAJOR.MINOR.PATCH-<alpha|beta|rc>n`, eg. `7.0.0-beta5`.*
+
+### Current Release Plan
+To be able to develop and improve Shopsys Framework we plan to release `MAJOR` versions almost quarterly, aiming to release a new `MAJOR` every 3-4 months.
+We expect this period to increase in the future to yearly releases.
+
+## Clarifications and Exceptions
+
+### Project-base Repository
+The [project-base repository](https://github.com/shopsys/project-base) is excepted for the BC promise because it is not meant to be depended upon or extended.
+It should be viewed as a template for your own e-commerce projects built on top of Shopsys Framework.
+
+Changes to the `project-base` may contain new features for front-end or examples of newly implemented features and configuration option.
+Follow the changes in the repository to see how working with Shopsys Framework changes between the versions and to keep in touch with best practices and recommendations.
+
+This means that the `project-base` should run with any higher version of Shopsys Framework, up to the next `MAJOR` version.
+
+*Note: The same holds true for the [demoshop repository](https://github.com/shopsys/demoshop) which is a complex example of an e-commerce project using a custom design and modifications.*
```

Comment:
I think demo shop shouldn't be mentioned here at all as it is not meant to be used at all.

Reply:
It's definitely meant to be used as an inspiration. It clarified things for me when I've compared the demoshop and project-base repositories in terms of upgradability and BC promice, so I've thought it might also help others.

Reply:
I read the BC promise from newbie point of view. And as a newbie I have no idea about demoshop. This note only shatters my focus to "what is demoshop?" "why should I care about it?" "why is it useful during reading about BC?"

Those are reasons why I prefer to remove this sentence.

But it leads me to another question **For whom is the BC promise?**
* newbies/people who are considering using SSFW - remove
* already implementing teams - keep
* both - remove

Reply:
I think it's for both and the contributors as well.

Reply:
in the end... why not, it's just one *Note*